VisibilityXds (and SpectrumXds): scan_number (int) => scan_name (str)

FedeMPouzols opened the pull request:
Resolves #333.

---
Note well:
```
This contribution is made under the current ALMA software agreements. 
(c) European Southern Observatory, 2024, 2025
Copyright by ESO (in the framework of the ALMA collaboration)
```

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/casangi/xradio?pullRequest=348) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/casangi/xradio?pullRequest=348) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/casangi/xradio?pullRequest=348) it.</sub>

FedeMPouzols commented:
Note that this will fail the tests with "pregenerated" vis.zarr datasets. I cannot upload to the dropbox folder, but I prepared this zip:  
[Antennae_North.cal.lsrk.split.ms.zip](https://github.com/user-attachments/files/18439721/Antennae_North.cal.lsrk.split.ms.zip)
This also needs to be changed for the S3 test I think (that one seems to have a "v5" which I'm not sure if implies some difference).

Jan-Willem commented:
@FedeMPouzols I also have some changes that will require a new Antennae_North.cal.lsrk.split. I will merge this branch into mine and generate the new Antennae_North.cal.lsrk.split.
